src/main/java/fr/insalyon/pldagile/view/maps/LineLayer.java

```diff
@@ -39,19 +46,49 @@ public void clearPoints() {
 
     @Override
     protected void layoutLayer() {
-        for (Pair<MapDestination, Line> candidate : lines) {
+        for (Pair<MapDestination, Pair<Line, ImageView>> candidate : lines) {
             MapDestination destination = candidate.getKey();
-            Line line = candidate.getValue();
+            Line line = candidate.getValue().getKey();
+            ImageView direction = candidate.getValue().getValue();
             MapPoint start = destination.getStart();
             MapPoint end = destination.getEnd();
 
+            // Draw the line
             Point2D startProjected = getMapPoint(start.getLatitude(), start.getLongitude());
             Point2D endProjected = getMapPoint(end.getLatitude(), end.getLongitude());
+
             line.setStartX(startProjected.getX());
             line.setStartY(startProjected.getY());
+
             line.setEndX(endProjected.getX());
             line.setEndY(endProjected.getY());
             line.setVisible(true);
+
+            // Draw the direction
+            // The icon's position is exactly at the middle of the line
+            Point2D midProjected = getMapPoint(start.getLatitude() + (end.getLatitude() - start.getLatitude()) / 2,
+                    start.getLongitude() + (end.getLongitude() - start.getLongitude()) / 2);
+            direction.setTranslateX(midProjected.getX());
+            direction.setTranslateY(midProjected.getY());
+
+            // Rotate the image of the arrow so it points in the direction of the end point
+            double rotateAngle = startProjected.angle(new Point2D(startProjected.getX() + 1, startProjected.getY()),
+                    endProjected);
+            /*
+             * Point2D.angle() will compute the smallest angle possible between the angle in
+             * the counter clockwise and the clockwise directions. However, the
+             * ImageView.rotate function rotates counter clockwise only. If the start point
+             * (startProjected) is below (has a lower y value) than the end point
+             * (endProjected), Point2D.angle(), will use the angle for the clockwise
+             * direction, thus, we must convert it to an angle in the counter clockwise
+             * direction.
```

Comment:
lmk if you have trouble understanding this sentence I wanted it to be clear in case someone else has to read over this